api/src/main/java/marquez/db/OpenLineageDao.java

```diff
@@ -135,6 +145,80 @@ default UpdateLineageRow updateMarquezModel(LineageEvent event, ObjectMapper map
     return updateLineageRow;
   }
 
+  default UpdateLineageRow updateMarquezModel(DatasetEvent event, ObjectMapper mapper) {
+    NamespaceDao namespaceDao = createNamespaceDao();
+    DatasetSymlinkDao datasetSymlinkDao = createDatasetSymlinkDao();
+    DatasetDao datasetDao = createDatasetDao();
+    SourceDao sourceDao = createSourceDao();
+    DatasetVersionDao datasetVersionDao = createDatasetVersionDao();
+    DatasetFieldDao datasetFieldDao = createDatasetFieldDao();
+    RunDao runDao = createRunDao();
+    DatasetFacetsDao datasetFacetsDao = createDatasetFacetsDao();
+    ColumnLineageDao columnLineageDao = createColumnLineageDao();
+
+    Instant now = event.getEventTime().withZoneSameInstant(ZoneId.of("UTC")).toInstant();
+
+    UpdateLineageRow bag = new UpdateLineageRow();
+    NamespaceRow namespace =
+        namespaceDao.upsertNamespaceRow(
+            UUID.randomUUID(),
+            now,
+            formatNamespaceName(event.getDataset().getNamespace()),
+            DEFAULT_NAMESPACE_OWNER);
+    bag.setNamespace(namespace);
+
+    Dataset dataset = event.getDataset();
+    List<DatasetRecord> datasetOutputs = new ArrayList<>();
+    DatasetRecord record =
+        upsertLineageDataset(
+            dataset,
+            now,
+            null,
+            false,
+            namespaceDao,
+            datasetSymlinkDao,
+            sourceDao,
+            datasetDao,
+            datasetVersionDao,
+            datasetFieldDao,
+            runDao,
+            columnLineageDao);
+    datasetOutputs.add(record);
+
+    // Facets ...
+    Optional.ofNullable(dataset.getFacets())
+        .ifPresent(
+            facets ->
+                datasetFacetsDao.insertDatasetFacetsFor(
+                    record.getDatasetRow().getUuid(),
+                    record.getDatasetVersionRow().getUuid(),
+                    null,
+                    now,
+                    null,
+                    facets));
+
+    // OutputFacets ...
```

Comment:
Do we not have InputFacets as well in this case?

Reply:
Neither `inputFacet` nor `outputFacets` should be  allowed for `DatasetEvent`.
We should have 3 methods: `insertDatasetFacets`, `insertInputDatasetFacets` and `insertOutputDatasetFacets`. 
I fixed this in the 5th commit.

Reply:
Ahh yes thanks for clarifying; was thinking about just this during my review 👀